### PR TITLE
build: automatically enable module dependencies

### PR DESCRIPTION
**Pull request by jonasnick**

### CODE REVIEW

File: configure.ac [hunk 1]
```diff
@@ -385,6 +385,10 @@ SECP_CFLAGS="$SECP_CFLAGS $WERROR_CFLAGS"
 ### Handle module options
 ###
 
+# Besides testing whether modules are enabled, the following code also enables
+# module dependencies. The order of the tests matters: the dependency must be
+# tested first.
+
```
Comment (real-or-random) [inline configure.ac:391]: nit: is the comment worth adding to upstream?

**Review by real-or-random — COMMENTED** [review]
Concept ACK

Though what I don't like too much about this approach that it silently enables options, even if explicitly disabled, e.g., in 
```
./configure --enable-experimental --enable-module-surjectionproof --disable-module-rangeproof
```
It may be better to add a notice "Automatically enabling xy" then. 
But yeah upstream does the same, so if anything, we should fix it there. (I'm fine either way)

---

The code further below that checks the errors can (should?) also be removed then.
 
---

Not this PR: 
The option listing in the end is incomplete. It lacks some modules, e.g., whitelist. Those should be added, and when we do this, we could also remove the (then redundant) list from the experimental warning. We should just keep the warning, but maybe move it further down, at the moment it's interleaved with some other configure output... I think this PR would be a good opportunity to fix this but feel free to ignore, we can also do it in a separate PR.

**Comment by jonasnick** [comment]
@real-or-random 

> Though what I don't like too much about this approach that it silently enables options, even if explicitly disabled

Oh, I missed that. I don't like that either. And yeah, if we keep it that way we can remove the error checking.

**Comment by jonasnick** [comment]
I don't see an straightforward way to error out if dependencies are explicitly disabled. If we fix the configure output with respect to the enabled modules maybe that's not so much of a problem.

File: configure.ac (hunk 1 above)
Comment (jonasnick) [reply configure.ac:391]: no imo, as there are comments already (what I removed here)

**Comment by jonasnick** [comment]
I removed the error checking and added two commits that improve the output.

**Comment by real-or-random** [comment]
> I don't see an straightforward way to error out if dependencies are explicitly disabled. If we fix the configure output with respect to the enabled modules maybe that's not so much of a problem.

Agreed, and PR is an improvement in the end, so we shouldn't hold it up.

**Review by real-or-random — APPROVED** [review]
utACK 9e58040be038670d62771e5a0dd4b79c932fffe0

**Comment by jonasnick** [comment]
@real-or-random  this seems to be a commit id belonging to a different PR :)

**Review by real-or-random — APPROVED** [review]
utACK 171b294a1c7a736c1b93fa194e3af90b625259fa

**Pull request merged**